flocx_market/api/app.py

```diff
@@ -39,7 +39,9 @@ def create_app(app_name):
         '/offer_contract_relationship',
         '/offer_contract_relationship/',
         '/offer_contract_relationship/'
-        '<string:marketplace_offer_id>/<string:contract_id>')
+        '<string:marketplace_offer_id>/<string:contract_id>',
+        '/offer_contract_relationship/'
+        '<string:marketplace_offer_id>/<string:contract_id>/<string:status>')
```

Comment:
Would it be possible to not create a new URL here, but to look at query parameters using request.args? Then you would just have to see if marketplace_offer_id, or contract_id, or status exists; and filter depending on whether they do.

Reply:
Oh, and correct me if I'm misunderstanding how this URL works!

Reply:
Hey, yes it's possible, I was just keeping the same pattern, but I can change it to that. I'll update it

Reply:
Hey. If the provider want to pull contracts by status, the marketplace has the contract_offer_relationship api to filter all contracts with status "unretrieved" and return only the contracts uuid. But the provider wants the contract information, like create time, price and so on. Can you do something like, when the contract_offer_relationship api has one parameter: status, then marketplace firstly queries the contract_offer_relationship table to get the contracts uuid, then queries contracts table to return these contract json data?

Reply:
I don't think that should be handled by the marketplace. After querying the offer_contract_relationships by status, the UUIDs of the corresponding contracts will be returned which can then be used to query the contracts. I think it's cleaner that way. 
What are your thoughts @tzumainn?

Reply:
@tzumainn, fixed up the URL thingy you asked for

Reply:
@DanNiESh @fvukelic I think a detailed query makes sense, but probably at a different URL; for example, Ironic has a nodes/ and a nodes/detail/ endpoint (as shown here: https://docs.openstack.org/api-ref/baremetal/#list-nodes-detailed)

I'd suggest a followup PR to implement the equivalent here. Otherwise, you're turning what could be one database query into 1 + <arbitrary number> queries.